src/parser/DASHTree.cpp

```diff
@@ -236,11 +236,14 @@ bool ParseContentProtection(const char** attr, DASHTree* dash)
 
 static void ReplacePlaceHolders(std::string& rep, const std::string& id, uint32_t bandwidth)
 {
-  std::string::size_type repPos = rep.find("$RepresentationID$");
-  if (repPos != std::string::npos)
+  while (1)
```

Comment:
on new changes on Nexus branch, is needed to clean a bit the code,
please replace existing method `replaceAll` in `helpers.cpp` with this:

```
int ReplaceAll(std::string &source, const std::string &oldStr, const std::string &newStr)
{
  if (oldStr.empty())
    return 0;

  int replacedChars = 0;
  size_t index = 0;

  while (index < source.size() && (index = source.find(oldStr, index)) != std::string::npos)
  {
    source.replace(index, oldStr.size(), newStr);
    index += newStr.size();
    replacedChars++;
  }

  return replacedChars;
}
```
then use `ReplaceAll` helper method to replace the "$RepresentationID$"
and use `replace` helper method to replace the first occurence of "$Bandwidth$"

Reply:
multiple $Bandwidth$ would be possible as well, so maybe a good idea to use the ReplaceAll for both?